### PR TITLE
improve the result of checking role name

### PR DESCRIPTION
**Pull request by xilabao**
Get all error messages before return.

### CODE REVIEW

**Comment by dims** [comment]
@k8s-bot ok to test

File: pkg/api/validation/path/name.go [hunk 1]
```diff
@@ -35,23 +35,31 @@ func IsValidPathSegmentName(name string) []string {
 		}
 	}
 
+	message := []string{}
```
Comment (liggitt) [inline pkg/api/validation/path/name.go]: declaring a slice var starts it out as nil, avoids the allocation in error-free cases, and removes the need to check for length:
```
var errors []string
...
for ... {
  errors = append(errors, ...)
}
return errors
```

File: pkg/api/validation/path/name.go [hunk 2]
```diff
@@ -35,23 +35,31 @@ func IsValidPathSegmentName(name string) []string {
...
+		return message
+	}
 
 	return nil
 }
 
 // IsValidPathSegmentPrefix validates the name can be used as a prefix for a name which will be encoded as a path segment
 // It does not check for exact matches with disallowed names, since an arbitrary suffix might make the name valid
 func IsValidPathSegmentPrefix(name string) []string {
+	message := []string{}
```
Comment (liggitt) [inline pkg/api/validation/path/name.go]: same here

**Review by liggitt — CHANGES_REQUESTED** [review]
small tweak on allocation and length check, looks good otherwise. a unit test would probably be good as well.

**Comment by xilabao** [comment]
@liggitt PTAL

File: pkg/api/validation/path/name_test.go [hunk 3]
```diff
@@ -130,3 +130,40 @@ func TestValidatePathSegmentName(t *testing.T) {
...
+func TestValidateWithMutliErrors(t *testing.T) {
+	testcases := map[string]struct {
+		Name        string
+		Prefix      bool
+		ExpectedMsg string
+	}{
+		"slash,percent": {
+			Name:        "foo//bar%",
+			Prefix:      false,
+			ExpectedMsg: "/%",
```
Comment (liggitt) [inline pkg/api/validation/path/name_test.go]: This is odd… I expected the list of expected messages

File: pkg/api/validation/path/name_test.go [hunk 4]
```diff
@@ -130,3 +130,40 @@ func TestValidatePathSegmentName(t *testing.T) {
 		}
 	}
 }
+
+func TestValidateWithMutliErrors(t *testing.T) {
```
Comment (liggitt) [inline pkg/api/validation/path/name_test.go]: Typo "Multi"

Comment (xilabao) [inline pkg/api/validation/path/name_test.go]: fixed

**Pull request merged**